lib/timeout.rb

```diff
@@ -132,6 +132,12 @@ def self.ensure_timeout_thread_created
         unless @timeout_thread and @timeout_thread.alive?
           @timeout_thread = create_timeout_thread
         end
+
+        # shut down timeout thread at exit
+        Kernel.at_exit do
+          @timeout_thread.kill
+          @timeout_thread.join rescue nil
+        end
```

Comment:
This should be move in the `unless` just above

Reply:
Copy/paste bug, I'll fix or replace with queue shutdown.